From 667f5f5f070702a590eff19fb9bf8893b20053c1 Mon Sep 17 00:00:00 2001
From: Jenny TC <jenny.tc@intel.com>
Date: Fri, 30 Jan 2015 09:34:38 +0530
Subject: [PATCH] power_supply: Use throttle state to detect charger prop
 change

Consider throttle state to detect change in charger properties.
So the algorithm would be invoked if the throttle state is changed
irrespective of other properties.

Change-Id: I40be71e1dff00ed8e896ad444e8739fc05443063
Signed-off-by: Jenny TC <jenny.tc@intel.com>
Signed-off-by: Ajay Thomas <ajay.thomas.david.rajamanickam@intel.com>
---
 drivers/power/power_supply_charger.c | 4 ++--
 drivers/power/power_supply_charger.h | 8 ++++----
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/power/power_supply_charger.c b/drivers/power/power_supply_charger.c
index b2b66e3..cab1cd4 100644
--- a/drivers/power/power_supply_charger.c
+++ b/drivers/power/power_supply_charger.c
@@ -306,6 +306,7 @@ static inline void get_cur_chrgr_prop(struct power_supply *psy,
 	chrgr_prop->cable = CABLE_TYPE(psy);
 	chrgr_prop->health = HEALTH(psy);
 	chrgr_prop->tstamp = get_jiffies_64();
+	chrgr_prop->throttle_state = CURRENT_THROTTLE_STATE(psy);
 
 }
 
@@ -370,6 +371,7 @@ update_props:
 	chrgr_cache->present = chrgr_prop_new->present;
 	chrgr_cache->cable = chrgr_prop_new->cable;
 	chrgr_cache->tstamp = chrgr_prop_new->tstamp;
+	chrgr_cache->throttle_state = chrgr_prop_new->throttle_state;
 }
 
 
@@ -445,7 +447,6 @@ update_props:
 	bat_cache->temperature = bat_prop_new->temperature;
 	bat_cache->status = bat_prop_new->status;
 	bat_cache->algo_stat = bat_prop_new->algo_stat;
-	bat_cache->throttle_state = bat_prop_new->throttle_state;
 }
 
 static inline int get_bat_prop_cache(struct power_supply *psy,
@@ -478,7 +479,6 @@ static inline void get_cur_bat_prop(struct power_supply *psy,
 	bat_prop->status = STATUS(psy);
 	bat_prop->health = HEALTH(psy);
 	bat_prop->tstamp = get_jiffies_64();
-	bat_prop->throttle_state = CURRENT_THROTTLE_STATE(psy);
 
 	/* Populate cached algo data to new profile */
 	ret = get_bat_prop_cache(psy, &bat_prop_cache);
diff --git a/drivers/power/power_supply_charger.h b/drivers/power/power_supply_charger.h
index 43d0d52..c7e2f18 100644
--- a/drivers/power/power_supply_charger.h
+++ b/drivers/power/power_supply_charger.h
@@ -28,7 +28,6 @@ struct batt_props {
 	unsigned long tstamp;
 	enum psy_algo_stat algo_stat;
 	int health;
-	int throttle_state;
 };
 
 struct charger_props {
@@ -40,6 +39,7 @@ struct charger_props {
 	bool online;
 	unsigned long cable;
 	unsigned long tstamp;
+	int throttle_state;
 };
 
 struct psy_batt_thresholds {
@@ -199,7 +199,8 @@ static inline int get_ps_int_property(struct power_supply *psy,
 	((cache_prop.online != prop.online) || \
 	(cache_prop.present != prop.present) || \
 	(cache_prop.is_charging != prop.is_charging) || \
-	(cache_prop.health != prop.health))
+	(cache_prop.health != prop.health) || \
+	(cache_prop.throttle_state != prop.throttle_state))
 
 #define IS_BAT_PROP_CHANGED(bat_prop, bat_cache)\
 	((bat_cache.voltage_now != bat_prop.voltage_now) || \
@@ -207,8 +208,7 @@ static inline int get_ps_int_property(struct power_supply *psy,
 	((bat_cache.current_now != bat_prop.current_now) || \
 	(bat_cache.voltage_now != bat_prop.voltage_now))) || \
 	(bat_cache.temperature != bat_prop.temperature) || \
-	(bat_cache.health != bat_prop.health) || \
-	(bat_cache.throttle_state != bat_prop.throttle_state))
+	(bat_cache.health != bat_prop.health))
 
 #define MAX_THROTTLE_STATE(psy)\
 		(get_ps_int_property(psy,\
-- 
1.9.1

